### PR TITLE
Add image sources doc

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

Add image sources doc for @dweomer 

#### Types of Changes ####

documentation

#### Verification ####

view the document

#### Linked Issues ####

* N/A

### CODE REVIEW

File: developer-docs/image_sources.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+| rancher-vsphere-csi | rancher/mirrored-k8scsi-csi-resizer | rancher/image-mirror |  | FALSE |
+| rancher-vsphere-csi | rancher/mirrored-k8scsi-livenessprobe | rancher/image-mirror |  | FALSE |
+|  |  |  |  |  |
+| harvester-cloud-provider | docker.io/rancher/harvester-cloud-provider | Harvester team |  | FALSE |
+|  |  |  |  |  |
+| harvester-csi-driver | docker.io/rancher/harvester-csi-driver | Harvester team |  | FALSE |
+| harvester-csi-driver | docker.io/rancher/longhornio-csi-attacher | Longhorn team |  | FALSE |
+| harvester-csi-driver | docker.io/rancher/longhornio-csi-node-driver-registrar | Longhorn team |  | FALSE |
+| harvester-csi-driver | docker.io/rancher/longhornio-csi-provisioner | Longhorn team |  | FALSE |
+| harvester-csi-driver | docker.io/rancher/longhornio-csi-resizer: | Longhorn team |  | FALSE |
```
Comment (dereknola) [inline developer-docs/image_sources.md:67]: Extra colon?

File: developer-docs/image_sources.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+This document lists all of the images used by RKE2 and any of its packaged components (Helm charts), along with info on where they are built and what other images they copy content from. If you find an image not on this list, please open a PR.
+
+| Chart | Image | Build Repo | Source Images | Hardened |
+| ----- | ----- | ---------- | ------------- | -------- |
+| CORE | rancher/hardened-kubernetes | rancher/image-build-kubernetes | ‣ ubi7/ubi-minimal<br>‣ rancher/hardened-build-base | TRUE |
+| CORE | rancher/rke2-runtime | rancher/rke2 | ‣ rancher/k3s<br>rancher/hardened-kubernetes<br>‣ rancher/hardened-containerd<br>‣ rancher/hardened-crictl<br>‣ rancher/hardened-runc | TRUE |
+| CORE FROM | rancher/hardened-containerd | rancher/image-build-containerd | ‣ ubi7/ubi-minimal<br>‣ rancher/hardened-build-base | TRUE |
```
Comment (dereknola) [inline developer-docs/image_sources.md:7]: Can you explain what CORE vs CORE FROM means?

**Review by dweomer — APPROVED** [review]
Thanks @brandond

File: developer-docs/image_sources.md (hunk 2 above)
Comment (brandond) [reply developer-docs/image_sources.md:9]: done

**Review by dereknola — APPROVED** [review]

**Pull request merged**